enclave-bridge/src/lib.rs

```diff
@@ -261,6 +312,49 @@ impl<T: Config> Pallet<T> {
 		)
 	}
 
+	pub fn get_maybe_updated_shard_config(
+		shard: ShardIdentifier,
+		apply_due_update: bool,
+	) -> Option<ShardConfig<T::AccountId>> {
+		let current_block_number = <frame_system::Pallet<T>>::block_number();
+		Self::shard_config(shard).map(|current| {
+			current.upgrade_at.filter(|&at| at <= current_block_number).map_or_else(
+				|| current.active_config.clone(),
+				|_| {
+					current.pending_upgrade.map_or(current.active_config.clone(), |due_update| {
+						if apply_due_update {
+							<ShardConfigRegistry<T>>::insert(
+								shard,
+								UpgradableShardConfig::from(due_update.clone()),
+							);
+						}
+						due_update
+					})
+				},
+			)
+		})
```

Comment:
With the change I suggested above, we could simplify this code a lot. What I don't like about the current approach:

* it mixes the concepts: config, and the update of the config
* the getter might return the new one or the old one
* the getter might mutate, given by a flag.